Visual Studio 2010

I am trying to run clcache with Visual Studio 2010.

I have tried putting it inthe PATH of my user, of the whole system and even replacing my cl.exe with this launcher  and speicyfing CLCACHE_CL to the correct compiler (renamed to cl.real.exe). My code let me specify where to find clcache.py, plus I am having problems with py2exe and my windows is x86_64.

In the current setup I have these variable set
```
CLCACHE_DIR = C:\clcache
CLCACHE_CL = C:\Program Files (x86)\Microsoft Visual Studio 10.0\VC\bin\cl.real.exe
CLCACHE_HARDLINK = 1
CLCACHE_LOG = 1
CLCACHE_PATH = C:\sources\clcache
```

```
#include <string>
#include <stdlib.h>

int main( int argc, char* argv[] )
{
    // Checking standard variables
    char *CLCACHE_PATH = getenv("CLCACHE_PATH");
    std::string cmd = "python "; 
    if (CLCACHE_PATH)
    {
        cmd += CLCACHE_PATH;
        cmd += '/';
    }
    cmd += "clcache.py ";
    for ( int i = 1; i < argc; ++i )
    {
        cmd += ' ';
        cmd += argv[i];
    }
    return system(cmd.c_str());
}
```

with the PATH modifications it wasn't being called, with this launcher I can manage to make it work if I manually call it, but visual studio will return me 

```
The system cannot execute the specified program.
C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\Platforms\Win32\Microsoft.Cpp.Win32.Targets(62,5): error MSB6006: "CL.exe" exited with code 1.
```

How can I setup Visual Studio 2010 to actually use clcache?

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -5,6 +5,8 @@ clcache changelog
 
  * Improvement: better protection against storing corrupt objects in the cache
    in case clcache is terminated in the middle of storing a new cache entry.
+ * Improvement: The README now explains an approach to making Visual Studio
+   pick up clcache.
  * Bugfix: Command files with multiple lines are now handled correctly.
  * Bugfix: Properly handle non-ASCII compiler output (GH #64)
 

--- a/README.asciidoc
+++ b/README.asciidoc
@@ -36,6 +36,17 @@ renamed and forward all the arguments to the real compiler executable.
 This way, simply running 'cl' will invoke the script instead of the real
 compiler.
 
+Installation For Visual Studio
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Some users have reported (see e.g. GitHub issue #18) that in order to make Visual Studio pick up `clcache`, the original compiler binary needs to be moved out of the
+way and replaced with the executable generated via the `py2exe` script:
+
+1. Rename `cl.exe` to e.g. `cl_original.exe`
+2. Rename `cl.exe.config` to e.g. `cl_original.exe.config`
+3. Copy `clcache.exe` to `cl.exe`
+4. Set `CLCACHE_CL` environment variable to point to `cl_original.exe`.
+
 Options
 ~~~~~~~
 

